### PR TITLE
Scheduler integration

### PR DESCRIPTION
**Pull request by shariq1501**
1. Added create schedule functionality

### CODE REVIEW

File: src/frontend/src/app/scheduled-process/scheduled-process.component.ts [hunk 1]
```diff
@@ -7,48 +7,119 @@ import { SchedulesService } from '../services/index';
...
+    let exclusive_state = <HTMLInputElement>document.getElementById("exclusive")
+    
+    if(this.selectedType == 2) {
+        var scheduler_day_field = <HTMLSelectElement>document.getElementById("day")
+        var scheduler_time_field = <HTMLInputElement>document.getElementById("time")
+        var day = scheduler_day_field.value
+        var time = scheduler_time_field.value
+        var scheduled_time = time != '' ? this.converTime(time):0
+    }
+    var repete_time = repeat_interval.value != '' ? this.converTime(repeat_interval.value):0
```
Comment (praveen-garg) [inline src/frontend/src/app/scheduled-process/scheduled-process.component.ts]: repeat_time

File: src/frontend/src/app/scheduled-process/scheduled-process.component.ts [hunk 2]
```diff
@@ -7,48 +7,119 @@ import { SchedulesService } from '../services/index';
...
+    this.invalidRepeat = this.between(repete_time) 
+    this.invalidTime =  this.between(scheduled_time)
+    if (this.invalidRepeat || this.invalidTime) {
```
Comment (praveen-garg) [inline src/frontend/src/app/scheduled-process/scheduled-process.component.ts]: I really doubt that we need these invalidRepeat, invalidTime var for `this.`
and method should be either not_bwtween or return NOT of current computation...

File: src/frontend/src/app/scheduled-process/scheduled-process.component.ts [hunk 3]
```diff
@@ -7,48 +7,119 @@ import { SchedulesService } from '../services/index';
...
+      error => { console.log("error", error) })
+  }
+
+  public between(x) {
+    // To check if Time in 00:00:01, 23:59:59 range
+    return x == 0 || x >= 86400
+  }
+
+  // Convert time in seconds 
+  converTime(timeValue) {
```
Comment (praveen-garg) [inline src/frontend/src/app/scheduled-process/scheduled-process.component.ts]: convertTimeToSec

File: src/frontend/src/app/scheduled-process/scheduled-process.component.ts [hunk 4]
```diff
@@ -7,48 +7,119 @@ import { SchedulesService } from '../services/index';
...
+    var repeatTime = timeValue.split(':')
+    var seconds = (+repeatTime[0]) * 60 * 60 + (+repeatTime[1]) * 60 + (+repeatTime[2])
```
Comment (praveen-garg) [inline src/frontend/src/app/scheduled-process/scheduled-process.component.ts]: curious to know use +, to make it positive integer?

File: src/frontend/src/app/scheduled-process/scheduled-process.component.ts [hunk 5]
```diff
@@ -7,48 +7,119 @@ import { SchedulesService } from '../services/index';
...
       data => {
         this.tasksData = data.tasks;
-        console.log("This is the tasks ",  data.tasks);
+        console.log("Tasks data", data.tasks)
       },
-      error => { console.log("error", error) });
+      error => { console.log("error", error) })
+  }
+
+  public setType(value){
```
Comment (praveen-garg) [inline src/frontend/src/app/scheduled-process/scheduled-process.component.ts]: not clear from the name for what ?

File: src/frontend/src/app/scheduled-process/scheduled-process.component.ts [hunk 6]
```diff
@@ -7,48 +7,119 @@ import { SchedulesService } from '../services/index';
...
+  }
+
+  public createSchedule() {
+    let schedule_name = <HTMLInputElement>document.getElementById("name")
+    let schedule_process = <HTMLSelectElement>document.getElementById("process")
+    let schedule_type = <HTMLSelectElement>document.getElementById("type")
+    let repeat_interval = <HTMLInputElement>document.getElementById("repeat")
+    let exclusive_state = <HTMLInputElement>document.getElementById("exclusive")
+    
+    if(this.selectedType == 2) {
```
Comment (praveen-garg) [inline src/frontend/src/app/scheduled-process/scheduled-process.component.ts]: selected what ? add a comment what is 0 or last index? and why  only 2

**Review by praveen-garg — CHANGES_REQUESTED** [review]

**Comment by shariq1501** [comment]
@praveen-garg conflicts resolved. You can review.

File: src/frontend/src/app/scheduled-process/scheduled-process.component.ts (hunk 4 above)
Comment (shariq1501) [reply src/frontend/src/app/scheduled-process/scheduled-process.component.ts]: >> `hms = '32:04:33';
  "32:04:33"
>> a = hms.split(':');
(3) ["32", "04", "33"]
>> (+a[0])
32`
To make it a integer value.

File: src/frontend/src/app/scheduled-process/scheduled-process.component.ts (hunk 2 above)
Comment (shariq1501) [reply src/frontend/src/app/scheduled-process/scheduled-process.component.ts]: invalidRepeat and invalidTime values are used for display time field range validation message  on html page only. That is the reason we need to make them class level var.

**Comment by shariq1501** [comment]
@praveen-garg Fixed as per feedback. Please have a look.

File: src/frontend/src/app/scheduled-process/scheduled-process.component.ts [hunk 7]
```diff
@@ -7,48 +7,134 @@ import { SchedulesService } from '../services/index';
...
+      var scheduler_time_field = <HTMLInputElement>document.getElementById("time")
+      var day = scheduler_day_field.value
+      var time = scheduler_time_field.value
+      var scheduled_time = time != '' ? this.converTimeToSec(time) : 0
+    }
+    var repeat_time = repeat_interval.value != '' ? this.converTimeToSec(repeat_interval.value) : 0
+    
+    // check if time is in valid range
+    this.invalidRepeat = this.not_between(repeat_time)
+    this.invalidTime = this.not_between(scheduled_time)
```
Comment (praveen-garg) [inline src/frontend/src/app/scheduled-process/scheduled-process.component.ts]: scheduled_time is in `if (this.selected_schedule_type == 2) {` is its not selected then undefined ?!!

**Review by praveen-garg — CHANGES_REQUESTED** [review]

**Comment by shariq1501** [comment]
@praveen-garg  Issue Fixed.

File: src/frontend/src/app/scheduled-process/scheduled-process.component.ts [hunk 8]
```diff
@@ -7,48 +7,139 @@ import { SchedulesService } from '../services/index';
...
+    this.selected_schedule_type = value;
+  }
+
+  public createSchedule() {
+    let schedule_name = <HTMLInputElement>document.getElementById("name")
+    let schedule_process = <HTMLSelectElement>document.getElementById("process")
+    let schedule_type = <HTMLSelectElement>document.getElementById("type")
+    let repeat_interval = <HTMLInputElement>document.getElementById("repeat")
+    let exclusive_state = <HTMLInputElement>document.getElementById("exclusive")
+
```
Comment (praveen-garg) [inline src/frontend/src/app/scheduled-process/scheduled-process.component.ts]:
```
let schedule_name = <HTMLInputElement>document.getElementById("name").value
let schedule_process = <HTMLSelectElement>document.getElementById("process").value
let schedule_type = <HTMLSelectElement>document.getElementById("type").value
let exclusive_state = <HTMLInputElement>document.getElementById("exclusive").checked
```

and then

let repeat_interval_fld = <HTMLInputElement>document.getElementById("repeat")
then use this to calculate var repeat_time

Also let's have var scheduled_time and var day here with default null value? and then in if `(this.selected_schedule_type == 2) {` check scheduled_time validity and assign this and the day from field

**Review by praveen-garg — APPROVED** [review]

**Pull request merged**